c/include/ml-api-common.h

```diff
@@ -405,6 +412,55 @@ const char * ml_error (void);
  */
 const char * ml_strerror (int errnum);
 
+/*************
+ * ML OPTION *
+ *************/
+
+/**
+ * @brief A handle of a ml-option instance.
+ * @since_tizen 7.0
+ */
+typedef void *ml_option_h;
+
+/**
+ * @brief Create ml-option instance.
+ * @since_tizen 7.0
+ * @param[out] option Newly created option handle is returned.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ */
+int ml_option_create (ml_option_h *option);
+
+/**
+ * @brief Destroy the ml-option instance.
+ * @details Note that, user should free the allocated value of ml-option in the case that destroy function is not given.
+ * @since_tizen 7.0
+ * @param[in] option The option handle to be destroyed.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ */
+int ml_option_destroy (ml_option_h option);
+
+/**
+ * @brief Set a new key-value in ml-option instance.
+ * @details key is a const string and value is a void pointer. Note that the value should be valid during single task and be freed after using unless proper destroy function is not given. When duplicated key is given, the corresponding value is updated with the new one.
+ * @since_tizen 7.0
+ * @param[in] option The handle of ml-option.
+ * @param[in] key The key to be set.
+ * @param[in] value The value to be set.
+ * @param[in] destroy The function to destroy the value. It is called when the ml-option instance is destroyed.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to start the pipeline.
+ */
+int ml_option_set (ml_option_h option, const char* key, void *value, ml_data_destroy_cb destroy);
```

Comment:
```
/**
 * @brief Sets a new key-value in ml-option instance.
 * @details key is a const string and value is a void pointer. Note that the value should be valid during single task and be freed after using unless proper destroy function is not given. When duplicated key is given, the corresponding value is updated with the new one.
 * @since_tizen 7.0
 * @param[in] option The handle of ml-option.
 * @param[in] key The key to be set.
 * @param[in] value The value to be set.
 * @param[in] destroy The function to destroy the value. It is called when the ml-option instance is destroyed.
 * @return @c 0 on success. Otherwise a negative error value.
 * @retval #ML_ERROR_NONE Successful.
 * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
 * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
 * @retval #ML_ERROR_STREAMS_PIPE Failed to start the pipeline.
 */
```